build/scripts/script.js

```diff
@@ -1031,3 +1106,87 @@ function getSearchIconElements() {
 function getSearchWrapperElement() {
   return document.getElementById("search-wrapper");
 }
+/**
+ * Creates an array of rendered elements based on the specified display level and item data.
+ *
+ * @param {Object} item - The item containing heading properties (h1, h2, h3).
+ * @param {number} displayFromLevel - The level from which to start rendering headings (1 to 3).
+ * @returns {Array} An array of elements created for the relevant heading levels.
+ */
+function createHeadingElements(item, displayFromLevel) {
+  const headingElements = [];
+  const headingLevelToRender = [];
+
+  if (displayFromLevel <= 1 && item.h1) {
+    headingLevelToRender.push("h1");
+  }
+
+  if (displayFromLevel <= 2 && item.h2) {
+    headingLevelToRender.push("h2");
+  }
+
+  if (displayFromLevel <= 3 && item.h3) {
+    headingLevelToRender.push("h3");
+  }
+
+  for (const headingLevel of headingLevelToRender) {
+    const element = createResultElement(item, headingLevel);
+    headingElements.push(element);
+  }
+  return headingElements;
+}
+
+function createResultElement(item, itemLevel) {
+  const links = searchIndexLinks[+item.id];
+  const contentDiv = document.createElement("div");
+  contentDiv.className = "search-result-item";
+  const locationDiv = document.createElement("div");
+  locationDiv.className = "search-result-location";
+
+  let href;
+  let textContent;
+  if (itemLevel === "h3") {
+    contentDiv.classList.add("search-result-item-is-h3");
+    if (links.anchorH3 !== undefined) {
+      href = rootUrl + "/" + links.file + "#" + links.anchorH3;
+    }
+    textContent = item.h3;
+  } else if (itemLevel === "h2") {
+    contentDiv.classList.add("search-result-item-is-h2");
+    if (links.anchorH2 !== undefined) {
+      href = rootUrl + "/" + links.file + "#" + links.anchorH2;
+    }
+    textContent = item.h2;
+  } else if (itemLevel === "h1") {
+    contentDiv.classList.add("search-result-item-is-h1");
+    if (links.anchorH1 !== undefined) {
+      href = rootUrl + "/" + links.file + "#" + links.anchorH1;
+    }
+    textContent = item.h1;
+  }
+
+  let anchorElement;
+  if (href) {
+    anchorElement = document.createElement("a");
+    anchorElement.href = href;
+  } else {
+    anchorElement = document.createElement("span");
+  }
+
+  anchorElement.textContent = textContent;
+  anchorElement.className = itemLevel;
+  locationDiv.appendChild(anchorElement);
+
+  const bodyDiv = document.createElement("div");
+  bodyDiv.className = "search-result-body";
+  let body = item.body ?? "";
+  if (body.length > 300) {
+    body = body.substring(0, 300) + "...";
+  }
+  bodyDiv.textContent = body;
```

Comment:
Just to check something: is it broken or ignored if we have just a body matching with no heading here?

Reply:
It will not be displayed